autodispose/src/main/java/com/uber/autodispose/ScopeUtil.java

```diff
@@ -117,12 +117,25 @@ private ScopeUtil() {
   public static <E> Maybe<LifecycleEndNotification> resolveScopeFromLifecycle(
       Observable<E> lifecycle,
       final E endEvent) {
+    Function<E, Boolean> equalityFunction;
+    if (endEvent instanceof Enum) {
+      // Match on enum ordinal in case we skip an event
+      final int targetOrdinal = ((Enum) endEvent).ordinal();
+      //noinspection unchecked
+      equalityFunction = (Function<E, Boolean>) new Function<Enum, Boolean>() {
```

Comment:
All java enums implement Comparable. You can use compareTo to get this behavior by default and allow consumers to provide their own if using custom enum events.